tools/autograd/gen_variable_type.py

```diff
@@ -474,6 +474,10 @@ def preprocess_nn_functions(declarations):
             declaration['derivative'] = declarations_by_name[base_name][0]['derivative']
             continue
 
+        if len(declarations_by_name[fwd_name]) != 1:
+            print(declarations_by_name[fwd_name])
+            import pdb
+            pdb.set_trace()
```

Comment:
Is this (and another set_trace below) some remainders of debugging?

Reply:
Yeah, this is absolutely not ready for merge. I spoke with @colesbury offline and there are some changes to the heuristic here we need to make